packages/ui/form/PhoneInput.tsx

```diff
@@ -18,11 +18,19 @@ export type PhoneInputProps = {
 };
 
 function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputProps) {
+  useEffect(() => {
+    if (!rest.value) {
+      return
+    }
+    const formattedValue = rest.value.trim().replace(/^\+?/, "+") 
+    onChange(formattedValue);
+  }, []);
+
   const defaultCountry = useDefaultCountry();
   return (
     <PhoneInput
       {...rest}
-      country={defaultCountry}
+      country={!rest.value ? defaultCountry : undefined}
```

Comment:
Have done this because if I haven't done this then it shows the default US country instead of selecting the country via itself.